src/EditorFeatures/Core/FindUsages/AbstractFindUsagesService.cs

```diff
@@ -17,20 +17,22 @@ internal abstract partial class AbstractFindUsagesService : IFindUsagesService
                 document, position, context.CancellationToken).ConfigureAwait(false);
             if (tuple == null)
             {
-                context.ReportMessage(EditorFeaturesResources.Cannot_navigate_to_the_symbol_under_the_caret);
+                await context.ReportMessageAsync(
+                    EditorFeaturesResources.Cannot_navigate_to_the_symbol_under_the_caret).ConfigureAwait(false);
                 return;
             }
 
             var message = tuple.Value.message;
 
             if (message != null)
             {
-                context.ReportMessage(message);
+                await context.ReportMessageAsync(message).ConfigureAwait(false);
                 return;
             }
 
-            context.SetSearchTitle(string.Format(EditorFeaturesResources._0_implementations,
-                FindUsagesHelpers.GetDisplayName(tuple.Value.symbol)));
+            await context.SetSearchTitleAsync(
```

Comment:
> This would cause it to operate as a non-blocking parallel operation 

To me this feels like a premature optimization.  I put in parallelization when it should lead to substantial improvements in user perceivable performance.  In this case, this is simply the natural transformation that comes from OOPifying a part of our API and is not up for consideration.

--

Or, in other words, i would only do the above if it was something i would be doing even if i were in proc.  The move to OOP doesn't really change that equation, though it does enable other parallism opportunities wrt not being as concerned with GC pauses (see the PR where we parallelize document search for NavigateTo).